fe/fe-core/src/main/java/org/apache/doris/mtmv/MTMVPlanUtil.java

```diff
@@ -40,28 +43,32 @@
 import org.apache.doris.nereids.trees.plans.visitor.TableCollector.TableCollectorContext;
 import org.apache.doris.qe.ConnectContext;
 import org.apache.doris.qe.SessionVariable;
+import org.apache.doris.qe.VariableMgr;
 
 import com.google.common.collect.Sets;
 
 import java.util.List;
+import java.util.Map;
 import java.util.Optional;
 import java.util.Set;
 
 public class MTMVPlanUtil {
 
-    public static ConnectContext createMTMVContext(MTMV mtmv) {
+    public static ConnectContext createMTMVContext(MTMV mtmv) throws DdlException {
         ConnectContext ctx = new ConnectContext();
         ctx.setEnv(Env.getCurrentEnv());
         ctx.setQualifiedUser(Auth.ADMIN_USER);
         ctx.setCurrentUserIdentity(UserIdentity.ADMIN);
         ctx.getState().reset();
         ctx.setThreadLocalInfo();
-        ctx.getSessionVariable().enableFallbackToOriginalPlanner = false;
-        ctx.getSessionVariable().enableNereidsDML = true;
-        ctx.getSessionVariable().allowModifyMaterializedViewData = true;
+        SessionVariable sessionVariable = ctx.getSessionVariable();
+        setPersonalizedSession(sessionVariable, mtmv);
```

Comment:
if user set  a sessions variable but is not exclusive by mtmv, what will happen?

Reply:
Any legitimate session can be added, which only affects the operation of the task